code/game/objects/items/weaponry.dm

```diff
@@ -505,8 +505,90 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	attack_verb_continuous = list("bludgeons", "whacks", "disciplines", "thrashes")
 	attack_verb_simple = list("bludgeon", "whack", "discipline", "thrash")
 
+/obj/item/cane/examine(mob/user, thats)
+	. = ..()
+	. += span_notice("This item can be used to support your weight, preventing limping from any broken bones on your legs you may have.")
+
+/obj/item/cane/equipped(mob/living/user, slot, initial)
+	. = ..()
+	movement_support_add(user, slot, initial)
+
+/obj/item/cane/dropped(mob/living/user, slot, initial)
+	. = ..()
+	movement_support_del(user, slot, initial)
+
+/obj/item/cane/proc/movement_support_add(mob/living/user, slot, initial)
+	if(!(slot & ITEM_SLOT_HANDS))
+		return
```

Comment:
You can make this a whole lot easier on yourself by only calling `movement_support_add` if it is equipped to a valid slot, rather than calling it regardless - 

That way you don't need to do the `. = ..() if(!.) return` song-and-dance for subtypes of `movement_support_add`